centaur/src/main/scala/centaur/CromwellTracker.scala

```diff
@@ -0,0 +1,60 @@
+package centaur
+
+import centaur.CromwellTracker._
+import cromwell.api.model.WorkflowMetadata
+import cromwell.core.{WorkflowProcessingEvent, WorkflowProcessingEvents}
+import cromwell.core.WorkflowProcessingEvents.DescriptionEventValue
+import io.circe.generic.semiauto._
+import io.circe.parser.parse
+import io.circe.syntax._
+import io.circe.{Decoder, Encoder}
+import org.apache.commons.math3.stat.inference.ChiSquareTest
+import org.slf4j.LoggerFactory
+
+import scala.language.postfixOps
+
+
+case class CromwellTracker(backendCount: Int, configuredSignificance: Double) {
+  val log = LoggerFactory.getLogger(getClass)
+  var counts: Map[String, Int] = Map()
+  def track(metadata: WorkflowMetadata): Unit = {
+
+    // Get all the workflow processing events for this workflow and pick out the "PickedUp" events. There can be multiple of
+    // these for a workflow if Cromwell was restarted while it was running.
+    val pickupEvents = parse(metadata.value) flatMap { _.hcursor.get[Array[WorkflowProcessingEvent]](WorkflowProcessingEvents.ProcessingEventsKey) }
+
+    pickupEvents match {
+      case Left(e) => throw new RuntimeException(e)
+      case Right(events) =>
+        events filter { _.description == DescriptionEventValue.PickedUp.value } foreach { e =>
+          synchronized {
+            counts = counts + (e.cromwellId -> (counts.getOrElse(e.cromwellId, 0) + 1))
+            log.info(s"added count for Cromwell id {}: {}", e.cromwellId, counts(e.cromwellId))
+          }
+        }
+    }
+  }
+
+  def assertHoricromtality(): Unit = {
+    counts foreach { case (id, count) => log.info(s"horicromtal count {}: {}", id, count) }
+
+    val runs = counts.values.sum
+    val expected: Array[Double] = (1 to backendCount) map { _ => runs.toDouble / backendCount } toArray
+    val actual: Array[Long] = counts.values map { _.toLong } toArray
+
+    val observedSignificance = new ChiSquareTest().chiSquareTest(expected, actual)
+    log.info(s"configured/observed horicromtal significance levels: {}/{}", configuredSignificance, observedSignificance)
```

Comment:
As is the `s""` isn't needed. Alternatively, after switching to Scala Logging, consider using the [`f""` interpolator](https://docs.scala-lang.org/overviews/core/string-interpolation.html#the-f-interpolator) to shorten the decimals to something like `%.4f` or so.

The `f""` could be used in the exception message below also.